internal/provider/manual/environ.go

```diff
@@ -110,7 +110,8 @@ func (e *manualEnviron) Create(context.ProviderCallContext, environs.CreateParam
 
 // Bootstrap is part of the Environ interface.
 func (e *manualEnviron) Bootstrap(ctx environs.BootstrapContext, callCtx context.ProviderCallContext, args environs.BootstrapParams) (*environs.BootstrapResult, error) {
-	provisioned, err := sshprovisioner.CheckProvisioned(e.host)
+	// On Bootstrap for manual providers, expect the provisioned check to be run as the Ubuntu user.
+	provisioned, err := sshprovisioner.CheckProvisioned(e.host, "", "")
```

Comment:
Passing in 2 empty strings is a bit magic.
Better to have 2 functions:
CheckProvisioned(host)
CheckProvisionedAsUser(host, user, key)

then use a private function that each calls to do thew work, or just have the CheckProvisioned call CheckProvisionedAsUser with the user param set to "ubuntu". Then we don't need the `if provisioningUser != ""` dance and can just set host to `<user>@host`

Reply:
Done! Good shout!